DANS-KNAW/dd-transfer-to-vault
# DD-1551 Add dependency health checks to dd-transfer-to-vault

Fixes DD-1551 Add dependency health checks to dd-transfer-to-vault

# Description of changes
Add two http connection `HealthCheck`s to dd-transfer-to-vault
- `Data-Vault-connection-check` checks connection to: `url: http://localhost:20365`
- `Vault-Catalog-connection-check` checks connection to:  `url: http://localhost:20305`

when the connection is failed:
-  an Error will be add to the log file. Example:
`ERROR [2024-07-14 17:00:52,660] nl.knaw.dans.ttv.health.HttpConnectionHealthCheck: Http health check against url=http://localhost:20365 failed with response=HttpHealthResponse{status=404, body='{"code":404,"message":"HTTP 404 Not Found"}'}`
- It will be mentioned in the admin healthy check monitor page. Example;

![image](https://github.com/user-attachments/assets/847fda61-2b95-42c2-8592-801caee3a595)

# How to test
navigate to the modules and run start scripts In the terminal tabs:
- `~/git/dans-core-systems/modules/dd-transfer-to-vault>  start-env.sh`
-  `~/git/dans-core-systems/modules/dd-transfer-to-vault>  start-hsqldb-server.sh`
- `~/git/dans-core-systems/modules/dd-transfer-to-vault>  start-service.sh`

- `~/git/dans-core-systems/modules/dd-vault-catalog> start-env.sh`
-  `~/git/dans-core-systems/modules/dd-vault-catalog> start-service.sh`
- `~/git/dans-core-systems/modules/dd-vault-catalog> start-hsqldb-server.sh`

- To run next service, see [`Docs/dev.md`](https://github.com/DANS-KNAW/dd-transfer-to-vault/blob/master/docs/dev.md). It is also possible to run following commands without those steps from documentation and see the connection error message in admin web page and `dd-transfer-to-vault` log file
- `~/git/dans-core-systems/modules/dd-data-vault> start-env.sh`
- `~/git/dans-core-systems/modules/dd-data-vault> start-hsqldb-server.sh`
- `~/git/dans-core-systems/modules/dd-data-vault> start-service.sh`

Monitoring:
- `dd-transfer-to-vault.log` file in the `data` folder of the `dd-transfer-to-vault` module
- [admin web page](http://localhost:8081/healthcheck?pretty=true)

# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/core-systems
